### PR TITLE
fix(manifest): remove namespace from disco svc manifest

### PR DESCRIPTION
**Pull request by iancoffey**
This seemingly errant namespace breaks my ability to stand up the cluster via the Makefile - the result is a consistent CrashLoopBackOff.

If this is not accidental, adding some words about how to avoid this failure would be useful.

### CODE REVIEW

**Comment by krancour** [comment]
I don't think this is accidental.  @technosophos and @mboersma and I had some discussion about keeping the dev/test manifests in each repo in the same namespace (`deis`) used by the official charts, such that a component one is hacking on is a viable drop-in replacement for a released component that was installed via helm.  As for why it causes an error, I'm not sure, but I would wonder if there's code somewhere in there that's assuming the pods are all running in the `default` namespace.

**Comment by iancoffey** [comment]
Gotcha, yeah it fails with an error about the deis name space not existing.

The main takeaway should be that the Usage instructions on this projects readme result in a broken etcd cluster on a fresh k8s installation, and if this doesnt get merged then something should happen to ensure those instructions result in a working setup.

**Comment by krancour** [comment]
@iancoffey that's interesting.  Where are you running said fresh k8s installation?

**Comment by iancoffey** [comment]
Im using a vanilla vagrant k8s environment, and I install etcd via this projects Usage instructions.

**Comment by krancour** [comment]
I was inquiring more about the k8s env itself.  e.g. kube-solo, micro-kube, etc.

The reason I ask is that which admission controls are enabled in your k8s cluster make a big difference here.  `NamespaceAutoProvision` will create namespaces just in time, while `NamespaceExists` will insist that namespaces are explicitly created in advance.

But it re-raises a good point (which is already addressed in the charts).  To run on the maximum number of clusters, we shouldn't make assumptions about how a cluster is configured.  And that means the `deis` namespace needs to be explicitly created.

This ties pretty directly into this issue (deis/charts#19) which is all about what the officially supported development workflow should be.

TL;DR for the issue referenced above, I'm of the opinion we should offer two options.  helm install deis first to start from _stock_ deis and then hack away _or_ BYO "deis," meaning a contributor who wants to hack on a component independent of the rest of the platform could create the `deis` namespace themselves, for instance, and I've proposed that we provide a make target that does so in order ot facilitate that with minimum friction.

**Comment by iancoffey** [comment]
@krancour Sure. I use the latest official [kubernetes release vagrant environment](https://github.com/kubernetes/kubernetes/blob/master/docs/getting-started-guides/vagrant.md). Yes, just creating the namespace is a valid fix, as long as its documented in this projects Usage.

Beyond that, I have no strong opinion about how this project should define its Usage instructions, beyond that it should work reliably and ideally reflect a common pattern across all the components.

**Comment by technosophos** [comment]
The long-term plan is to focus more on the Helm-based install (which will have configurable namespaces in the future), and for the charts here to be sort of "starting points" that people hack on when they don't want to use Helm for some reason. I don't know if that's a particularly compelling story. Nor do I know whether that gives a really solid "right" answer to your question @iancoffey. I think the namespaces are showing up in all of the repos because it's easier for Deis devs to work on things using the Makefile workflow.

**Comment by krancour** [comment]
> I think the namespaces are showing up in all of the repos because it's easier for Deis devs to work on things using the Makefile workflow.

I was seeing it a bit differently, @technosophos.  Currently, the charts are putting things in the `deis` namespace.  I know we expect this to be configurable in the future, but it's not currently.  So, in my head, the reason the `deis` namespace is starting to showing up in manifests is so that any given component "x", as described by such a manifest is an adequate drop-in replacement for for component "x" in the `deis` namespace, as described by the helm charts.

I strongly believe that stock helm install is going to be the most common jumping off point for anyone who wants to hack on any given component.  And if not, then it's (partially) on them to satisfy whatever dependencies may be missing to support hacking any given component in isolation.  ("Partially" because we could add a make target to facilitate stubbing out the dependencies.)

**Comment by technosophos** [comment]
I am marking this LGTM because I think removing the namespace shouldn't introduce any problems to deis users (who already set `--namespace` on the cli), but makes this more flexible for others.

**Comment by arschles** [comment]
LGTM2 for the same reasons that @technosophos said. also, if we're going to continue to have a copy of the manifests in our repos, I agree that they should be oriented toward general purpose development & deployment
